addons/spreadsheet_dashboard/models/spreadsheet_dashboard_share.py

```diff
@@ -0,0 +1,46 @@
+import base64
+import uuid
+
+from odoo import models, fields, api, exceptions
+from odoo.tools import consteq
+
+class SpreadsheetDashboardShare(models.Model):
+    _name = 'spreadsheet.dashboard.share'
+    _inherit = 'spreadsheet.mixin'
+    _description = 'Copy of a shared dashboard'
+
+    dashboard_id = fields.Many2one('spreadsheet.dashboard', required=True, ondelete='cascade')
+    excel_export = fields.Binary()
+    access_token = fields.Char(required=True, default=lambda x: str(uuid.uuid4()), groups="base.group_user")
+    full_url = fields.Char(string="URL", compute='_compute_full_url')
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if "excel_files" in vals:
+                excel_zip = self._zip_xslx_files(
+                    vals["excel_files"]
+                )
+                del vals["excel_files"]
+                vals["excel_export"] = base64.b64encode(excel_zip)
```

Comment:
I do not like the idea of having `"excel_files"` which is not a field in the `create` `vals` dictionary. It's supposed to only have valid fields values in there.

Can you move this zip logic inside `action_get_share_url`, before calling `create` (`action_get_share_url` is our own custom method, so `vals` doesn't have any existing API we should follow)